SignalMessaging/categories/UIColor+OWS.m

```diff
@@ -273,26 +273,46 @@ + (UIColor *)ows_darkSkyBlueColor
     return colorMap;
 }
 
-+ (NSArray<NSString *> *)ows_conversationColorNames
++ (NSDictionary<NSString *, UIColor *> *)ows_conversationColorMapDark
 {
-    return self.ows_conversationColorMap.allKeys;
+    static NSDictionary<NSString *, UIColor *> *colorMap;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        colorMap = @{
+            @"red" : self.ows_red700Color,
```

Comment:
Shouldn't there be 12 colors?

I also suspect that instead of passing around string arguments, we'd benefit from passing around strongly typed ConversationColor enum members, and have the ability to map it to/from strings for the purposes of serialization.

Reply:
This is the PR that actually updates these constants: https://github.com/signalapp/Signal-iOS/pull/3906

Yes, we could definitely use an enum.